Git commit 1f0f210ad1d8286ba7a8254664489f922cbba0c3 by Noah Davis.
Committed on 31/08/2020 at 12:46.
Pushed by ndavis into branch 'master'.

Don't use parent height/width for implicit ToolSeparator sizing

I realized this was pretty much all wrong, so I rewrote the whole thing with a structure inspired by the Default QQC2 theme.

Padding property was removed since it was originally added to give the
separator back the padding that it lost by having the same height as the
parent.

diff --git org.kde.desktop/ToolSeparator.qml org.kde.desktop/ToolSeparator.qml
index 197cd46..80d172f 100644
--- org.kde.desktop/ToolSeparator.qml
+++ org.kde.desktop/ToolSeparator.qml
@@ -12,20 +12,14 @@ import org.kde.kirigami 2.4 as Kirigami
 T.ToolSeparator {
     id: controlRoot
 
-    topPadding: 0
-    bottomPadding: 0
-    leftPadding: Kirigami.Units.smallSpacing
-    rightPadding: Kirigami.Units.smallSpacing
+    implicitWidth: Math.max(implicitBackgroundWidth + leftInset + rightInset,
+                            implicitContentWidth + leftPadding + rightPadding)
+    implicitHeight: Math.max(implicitBackgroundHeight + topInset + bottomInset,
+                             implicitContentHeight + topPadding + bottomPadding)
 
-    implicitWidth: separator.width + controlRoot.leftPadding + controlRoot.rightPadding
-    implicitHeight: parent.height
-
-    background: Kirigami.Separator {
-        id: separator
-        anchors {
-            top: controlRoot.top
-            bottom: controlRoot.bottom
-            horizontalCenter: controlRoot.horizontalCenter
-        }
+    contentItem: Kirigami.Separator {
+        // implicitHeight is the same as ToolBar implicitHeight minus ToolBar padding if not horizontal
+        implicitHeight: horizontal ? Math.floor(Kirigami.Units.devicePixelRatio) : 40 - (Kirigami.Units.smallSpacing * 2)
+        implicitWidth: horizontal ? 40 - (Kirigami.Units.smallSpacing * 2) : Math.floor(Kirigami.Units.devicePixelRatio)
     }
 }
